flutter_mobx/lib/src/observer.dart

```diff
@@ -16,39 +16,59 @@ bool debugAddStackTraceInObserverName = true;
 /// See also:
 ///
 /// - [Builder], which is the same thing but for [StatelessWidget] instead.
-class Observer extends StatelessObserverWidget
-    // Implements Builder to import the documentation of `builder`
-    implements
-        // ignore: avoid_implementing_value_types
-        Builder {
+class Observer extends StatelessObserverWidget {
   // ignore: prefer_const_constructors_in_immutables
   Observer({
     Key? key,
     required this.builder,
     String? name,
     bool? warnWhenNoObservables,
   })  : debugConstructingStackFrame = debugFindConstructingStackFrame(),
+        builderOptimized = null,
+        child = null,
+        assert(builder != null),
         super(
           key: key,
           name: name,
           warnWhenNoObservables: warnWhenNoObservables,
         );
 
-  @override
-  final WidgetBuilder builder;
+  /// Observer which excludes the child branch
+  // ignore: prefer_const_constructors_in_immutables
+  Observer.optimized({
```

Comment:
We should call this `withChild` to make the intent clear. The idea is to only build parts that are actually changing and keeping rest of the tree intact. I think you would also need a `withChildren` variant as well which could be a `Map<String, Widget>`?. This variant will help in plugging in parts of the tree which are not changing. This is useful when you have more than one child.